### PR TITLE
[MM-45749] Support for advanced ICE server configs and TURN credentials

### PR DESCRIPTION
**Pull request by streamer45**
#### Summary

Porting https://github.com/mattermost/mattermost-mobile/pull/6371 and https://github.com/mattermost/mattermost-mobile/pull/6404 to v2.

#### Ticket Link

https://mattermost.atlassian.net/browse/MM-45749
https://mattermost.atlassian.net/browse/MM-45751

#### Release Note

```release-note
Added support for advanced ICE server configs and short-lived TURN credentials.
```

### CODE REVIEW

File: app/products/calls/types/calls.ts
Comment (streamer45) [inline app/products/calls/types/calls.ts:109]: @cpoile I was slightly confused by this given it's currently a superset of `ServerCallsConfig`. Is there anything that can be done to avoid the redundancy here?

File: app/products/calls/connection/connection.ts
Comment (enahum) [inline app/products/calls/connection/connection.ts:14]:
```suggestion
import type {CallsConnection} from '@calls/types/calls';
```

File: app/products/calls/client/rest.ts
Comment (enahum) [inline app/products/calls/client/rest.ts:4]:
```suggestion
import type {ServerChannelState, ServerCallsConfig, ICEServersConfigs} from '@calls/types/calls';
```

File: app/products/calls/connection/simple-peer.ts
Comment (enahum) [inline app/products/calls/connection/simple-peer.ts:24]:
```suggestion
import type {ICEServersConfigs} from '@calls/types/calls';
```

File: app/products/calls/types/calls.ts
Comment (enahum) [inline app/products/calls/types/calls.ts:4]: are this types? if so maybe we should import them as such?

File: app/products/calls/utils.ts
Comment (enahum) [inline app/products/calls/utils.ts:7]:
```suggestion
import type {CallParticipant, ServerCallsConfig} from '@calls/types/calls';
```

**Review by enahum — CHANGES_REQUESTED** [review]
Overall looks good, just needs some updates to the imports of type definitions ;)

File: app/products/calls/types/calls.ts
Comment (cpoile) [reply app/products/calls/types/calls.ts]: Yep, good point. We can use an intersection type to remove the redundancy.

**Review by cpoile — APPROVED** [review]

**Comment by cpoile** [comment]
We should wait for this to be tested on iOS, like #6543

**Review by enahum — APPROVED** [review]

**Comment by enahum** [comment]
> We should wait for this to be tested on iOS, like #6543

@cpoile added labels on both PR's to wait for next release... if you happen to test it before today's build and want to merge them, feel free to do so.

**Comment by cpoile** [comment]
Tested on iOS, good to go.

**Pull request merged**
